File: Ide.php
<!DOCTYPE html>
<html>
<head>
  <title>Calculator</title>
</head>
<body>
  <input id="display" type="text"><br>
  <button onclick="buttonClick('7')">7</button>
  <button onclick="buttonClick('8')">8</button>
  <button onclick="buttonClick('9')">9</button>
  <button onclick="buttonClick('+')">+</button><br>
  <button onclick="buttonClick('4')">4</button>
  <button onclick="buttonClick('5')">5</button>
  <button onclick="buttonClick('6')">6</button>
  <button onclick="buttonClick('-')">-</button><br>
  <button onclick="buttonClick('1')">1</button>
  <button onclick="buttonClick('2')">2</button>
  <button onclick="buttonClick('3')">3</button>
  <button onclick="buttonClick('*')">*</button><br>
  <button onclick="buttonClick('0')">0</button>
  <button onclick="buttonClick('.')">.</button>
  <button onclick="buttonClick('=')">=</button>
  <button onclick="buttonClick('/')">/</button><br>
  <button onclick="clearCalculator()">Clear</button>
  <script src="calculator.js"></script>
</body>
</html>

File: Ide.java
// Define variables for the calculator
let num1 = 0;
let num2 = 0;
let operator = "";

// Function to perform arithmetic operations
function calculate() {
  let result = 0;
  switch(operator) {
    case "+":
      result = num1 + num2;
      break;
    case "-":
      result = num1 - num2;
      break;
    case "*":
      result = num1 * num2;
      break;
    case "/":
      result = num1 / num2;
      break;
    default:
      result = 0;
  }
  return result;
}

// Function to handle button clicks
function buttonClick(value) {
  if (isNaN(value)) {
    // If value is an operator
    if (value === "=") {
      // Calculate result and display it
      num2 = parseFloat(document.getElementById("display").value);
      document.getElementById("display").value = calculate();
      num1 = 0;
      num2 = 0;
      operator = "";
    } else {
      // Set operator and save first number
      operator = value;
      num1 = parseFloat(document.getElementById("display").value);
      document.getElementById("display").value = "";
    }
  } else {
    // If value is a number, add it to the display
    document.getElementById("display").value += value;
  }
}

// Function to clear the calculator
function clearCalculator() {
  num1 = 0;
  num2 = 0;
  operator = "";
  document.getElementById("display").value = "";
}
